### DISCUSSION TITLE
很好奇如果把线程数设置到超过1000会怎样？

### DISCUSSION THREAD

**Discussion by Idealisten** [XIU2/CloudflareSpeedTest#437] [提问]
我设置到2000好像会在测延迟阶段直接跳过前面90%的IP。
另外现在家用CPU线程数一般都不超过100吧，为啥这边默认设置200呢？设置的线程数超过CPU实际线程数会怎样呢？

**Comment by XIU2** [comment]
最初默认值是 500，大部分人使用起来是没有问题的，只是部分网络设备性能较弱的、或运营商对连接数限制苛刻的、或在微小型设备上运行的会出现网络卡顿甚至断网的情况，所以我干脆给调低了。

另外，各个应用程序的 线程 和 CPU 的内核、逻辑处理器不是一个概念，不要混淆一谈。
你就没想过，很多下载工具动不动就是 16、32 线程下载的，而很多人的 CPU 还都是 12 线程的，你猜为啥还能正常运行？

另外，CloudflareST 的 -n 200 延迟测速线程，指的是同时发起 200 个 TCPing/HTTPing 链接进行延迟测试，设置的太高，会占用过多网络资源，互相挤占网络资源的话就会导致测速结果失准（网络卡顿），另外也容易导致触发运营商的连接数限制（各地各运营商都不一样），直接给你临时断网什么的。
